File: solutions/javascript/RemoveDuplicatesfromSortedList/RemoveDuplicatesfromSortedList.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    let new_head = new ListNode('head');
    new_head.next = head;
    while (head) {
        if (head.next && head.next.val === head.val) {
            head.next = head.next.next;
        } else {
            head = head.next;
        }
    }
    return new_head.next;
};

File: solutions/java/RemoveDuplicatesfromSortedList/RemoveDuplicatesfromSortedList.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode nh = new ListNode(0);
        nh.next = head;
        while (null != head) {
            if (null != head.next && head.next.val == head.val) {
                head.next = head.next.next;
            } else {
                head = head.next;
            }
        }
        return nh.next;
    }
}